### ISSUE TITLE
Error while building GCC5 on Mac OS X 10.13.6

### ISSUE THREAD

**Issue by x-ii-ii-x** [fink/fink-distributions#470]
Good Day the Fink Team,

While installing the Bundle KDE4 Mac, an error was thrown while building GCC 5.5:

```
/bin/sh ../libtool --tag=CXX   --mode=compile /sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/./gcc/xgcc -shared-libgcc -B/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/./gcc -nostdinc++ -L/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/x86_64-apple-darwin17.7.0/libstdc++-v3/src -L/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/x86_64-apple-darwin17.7.0/libstdc++-v3/src/.libs -L/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/x86_64-apple-darwin17.7.0/libstdc++-v3/libsupc++/.libs -B/sw/lib/gcc5/x86_64-apple-darwin17.7.0/bin/ -B/sw/lib/gcc5/x86_64-apple-darwin17.7.0/lib/ -isystem /sw/lib/gcc5/x86_64-apple-darwin17.7.0/include -isystem /sw/lib/gcc5/x86_64-apple-darwin17.7.0/sys-include    -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DASAN_HAS_EXCEPTIONS=1 -DASAN_FLEXIBLE_MAPPING_AND_OFFSET=0 -DASAN_NEEDS_SEGV=1 -DMAC_INTERPOSE_FUNCTIONS -DMISSING_BLOCKS_SUPPORT -I. -I../../../../gcc-5.5.0/libsanitizer/asan -I..  -I ../../../../gcc-5.5.0/libsanitizer/include -I ../../../../gcc-5.5.0/libsanitizer  -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros -fno-ipa-icf -I../../libstdc++-v3/include     -I../../libstdc++-v3/include/x86_64-apple-darwin17.7.0     -I../../../../gcc-5.5.0/libsanitizer/../libstdc++-v3/libsupc++ -std=gnu++11 -g -O2 -MT asan_malloc_mac.lo -MD -MP -MF .deps/asan_malloc_mac.Tpo -c -o asan_malloc_mac.lo ../../../../gcc-5.5.0/libsanitizer/asan/asan_malloc_mac.cc
libtool: compile:  /sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/./gcc/xgcc -shared-libgcc -B/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/./gcc -nostdinc++ -L/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/x86_64-apple-darwin17.7.0/libstdc++-v3/src -L/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/x86_64-apple-darwin17.7.0/libstdc++-v3/src/.libs -L/sw/src/fink.build/gcc5-5.5.0-1/darwin_objdir/x86_64-apple-darwin17.7.0/libstdc++-v3/libsupc++/.libs -B/sw/lib/gcc5/x86_64-apple-darwin17.7.0/bin/ -B/sw/lib/gcc5/x86_64-apple-darwin17.7.0/lib/ -isystem /sw/lib/gcc5/x86_64-apple-darwin17.7.0/include -isystem /sw/lib/gcc5/x86_64-apple-darwin17.7.0/sys-include -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DASAN_HAS_EXCEPTIONS=1 -DASAN_FLEXIBLE_MAPPING_AND_OFFSET=0 -DASAN_NEEDS_SEGV=1 -DMAC_INTERPOSE_FUNCTIONS -DMISSING_BLOCKS_SUPPORT -I. -I../../../../gcc-5.5.0/libsanitizer/asan -I.. -I ../../../../gcc-5.5.0/libsanitizer/include -I ../../../../gcc-5.5.0/libsanitizer -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros -fno-ipa-icf -I../../libstdc++-v3/include -I../../libstdc++-v3/include/x86_64-apple-darwin17.7.0 -I../../../../gcc-5.5.0/libsanitizer/../libstdc++-v3/libsupc++ -std=gnu++11 -g -O2 -MT asan_malloc_mac.lo -MD -MP -MF .deps/asan_malloc_mac.Tpo -c ../../../../gcc-5.5.0/libsanitizer/asan/asan_malloc_mac.cc  -fno-common -DPIC -o .libs/asan_malloc_mac.o
In file included from ../../../../gcc-5.5.0/libsanitizer/asan/asan_malloc_mac.cc:17:0:
/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:663:61: error: expected initializer before 'API_AVAILABLE'
 CFTypeRef CFAutorelease(CFTypeRef CF_RELEASES_ARGUMENT arg) API_AVAILABLE(macos(10.9), ios(7.0), watchos(2.0), tvos(9.0));
                                                             ^

make[4]: *** [asan_malloc_mac.lo] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-target-libsanitizer] Error 2
make: *** [bootstrap-lean] Error 2
### execution of /tmp/fink.IuMo7 failed, exit code 2
### execution of /tmp/fink.4lhxU failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/sw/bin/dpkg-lockwait -r fink-buildlock-gcc5-5.5.0-1
(Reading database ... 299350 files and directories currently installed.)
Removing fink-buildlock-gcc5-5.5.0-1 ...
Failed: phase compiling: gcc5-5.5.0-1 failed
Package manager version: 0.45.0
Distribution version: selfupdate-rsync Thu Sep 26 23:07:13 2019, 10.13, x86_64
Trees: local/main stable/main
Xcode.app: 10.1
Xcode command-line tools: 10.1.0.0.1.1539992718
Max. Fink build jobs:  1
```

Digging with this error, it looks like a known bug:
- https://trac.macports.org/ticket/56502
- https://github.com/macports/macports-ports/commit/783216ed1e2475122420d02726c47da27f46292b

I'm gonna give it a try

**Comment by nieder** [comment]
This should be fixed by the patchfile `gcc5-10.13_headers.patch` that's currently only applied to 10.14 and up. Most probably it wasn't applied to 10.13 originally because 10.13 was on Xcode9 at the time and this is an xcode10 issue.

Time to go conditionalize this.

**Comment by x-ii-ii-x** [comment]
I didn't know about the XCode issue so I already created by hand the patch before your replied to me. That shouldn't be a problem because it's the same as `gcc5-10.13_headers.patch` ^^.
The build was successfull ! ! ! Restarted the bundle-kde4-mac, keep in touch.
